core/domain/classifier_services.py

```diff
@@ -127,6 +127,70 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+
+
+def add_to_training_queue(exploration):
+    """Checks all states of the exploration for re-training conditions and
+    creates ClassifierTrainingJobModel instances in the datastore
+    accordingly.
+
+    Args:
+        exploration: Exploration. Domain object for an exploration.
+    """
+    states = exploration.states
+    for state_name in states:
+        state = states[state_name]
+        if not state.can_undergo_classification():
+            continue
+        exp_id = exploration.id
+        exp_version = exploration.version
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        classifier = get_classifier_from_exploration_attributes(
+            exp_id, exp_version, state_name)
+        if classifier:
+            if not check_re_training_conditions(state, classifier.id):
+                create_classifier_exploration_mapping(exp_id, exp_version+1,
+                                                      state_name, classifier.id)
+                continue
+        create_classifier_training_job(algorithm_id, interaction_id, exp_id,
```

Comment:
@pranavsid98 check this [sections](https://docs.google.com/document/d/1kACmivEvZVlTYBdqxy-mmdTi3sg_NbkZdS10T-ya-pg/edit#)'s Question and Answer part in bottom.

Consider following scenario:
* creator modifies training data and a new job has been created as a result.
* creator again modifies data before new job goes to VM for training.

So if we fetch training data at the time of sending job to VM then we always have latest training data. Otherwise you would have to modify the pending job's training data.

The logic of jobs' is that only create a job if there is no pending job for same state and same exploration but if there is then don't do anything.

I am yet to see your PR but this is the original design we considered for jobs and training data.